destination/parse.go

```diff
@@ -164,45 +164,46 @@ func (d deviceFinder) parseDeviceList() (*deviceList, error) {
 
 		for _, deviceList := range list.Devices {
 			for _, device := range deviceList {
-				if device.IsAvailable {
-					return nil
+				if !device.IsAvailable {
+					d.logger.Warnf("device %s is unavailable: %s", device.Name, device.AvailabilityError)
 				}
 			}
 		}
 
-		return fmt.Errorf("no device is available")
+		return nil
 	}); err != nil {
 		return &deviceList{}, err
 	}
 
 	return &list, nil
 }
 
-func (d deviceFinder) filterDeviceList(wantedDevice Simulator) (Device, error) {
+func (d deviceFinder) deviceForDestination(wantedDestination Simulator) (Device, error) {
 	if d.list == nil {
 		return Device{}, fmt.Errorf("inconsistent state in filterDeviceList: device list should be parsed")
 	}
 
-	wantedPlatform := wantedDevice.Platform
-	wantedDevice.Platform = strings.TrimSuffix(wantedDevice.Platform, " Simulator")
+	wantedPlatform := wantedDestination.Platform
+	wantedDestination.Platform = strings.TrimSuffix(wantedDestination.Platform, " Simulator")
 
-	runtime, err := d.filterRuntime(wantedDevice)
+	runtime, err := d.runtimeForPlatformVersion(wantedDestination.Platform, wantedDestination.OS)
 	if err != nil {
 		return Device{}, err
 	}
 	runtimeID := runtime.Identifier
 
-	devices, ok := d.list.Devices[runtimeID]
+	devicesOfRuntime, ok := d.list.Devices[runtimeID]
 	if !ok {
-		return Device{}, fmt.Errorf("runtime (%s) not found", runtimeID)
+		return Device{}, fmt.Errorf("no device exists for runtime %s", runtime.Name)
 	}
 
 	// As the name of the device matches the device type ('iPhone 11') for factory created devices, look up device by name.
-	// If the default name is required and already created will use that.
-	for _, device := range devices {
-		if device.Name == wantedDevice.Name {
+	// If the default name is required and already created, it will use that.
+	for _, device := range devicesOfRuntime {
+		if device.Name == wantedDestination.Name {
 			if !device.IsAvailable {
-				return Device{}, fmt.Errorf("device (%s) with runtime OS (%s) is unavailable: %s", wantedDevice.Name, runtime.Version, device.AvailabilityError)
+				d.logger.Warnf("device %s for %s is unavailable: %s", device.Name, runtime.Name, device.AvailabilityError)
+				continue
```

Comment:
It's possible that another iteration of the loop will find a matching device. And even if there is no match, there is a fallback logic down below.